src/mod_push_service_mongoosepush.erl

```diff
@@ -156,29 +156,28 @@ parse_api_version(_) ->
     {error, not_supported}.
 
 %% Create notification for API v2 and v3
-make_notification(Notification, Options = #{<<"silent">> := <<"true">>}) ->
-    MessageCount = binary_to_integer(maps:get(<<"message-count">>, Notification)),
-    {ok, #{
-        service => maps:get(<<"service">>, Options),
-        mode => maps:get(<<"mode">>, Options, <<"prod">>),
-        topic => maps:get(<<"topic">>, Options, null),
-        data => Notification#{<<"message-count">> => MessageCount},
-        priority => maps:get(<<"priority">>, Options, normal)
-    }};
 make_notification(Notification, Options) ->
-    {ok, #{
-        service => maps:get(<<"service">>, Options),
-        mode => maps:get(<<"mode">>, Options, <<"prod">>),
-        alert => #{
-            body => maps:get(<<"last-message-body">>, Notification),
-            title => maps:get(<<"last-message-sender">>, Notification),
-            tag => maps:get(<<"last-message-sender">>, Notification),
-            badge => binary_to_integer(maps:get(<<"message-count">>, Notification)),
-            click_action => maps:get(<<"click_action">>, Options, null)
-        },
-        topic => maps:get(<<"topic">>, Options, null),
-        priority => maps:get(<<"priority">>, Options, normal)
-    }}.
+    RequiredParameters = #{service => maps:get(<<"service">>, Options)},
+    OptionalKeys = [<<"mode">>, <<"priority">>, <<"time_to_live">>,
```

Comment:
Maybe we should put this list somewhere more visible, like a definition at the top of the file?

Reply:
Agree with that one

Reply:
If definition is used just once, i prefer to have to have it at the place where it used.